cartesius/data.py

```diff
@@ -128,17 +144,21 @@ def _gen_poly(self, x_ctr, y_ctr, avg_radius, irregularity, spikeyness, n_vert):
         # Now generate the points
         angle = random.uniform(0, 2 * math.pi)
         angles = np.cumsum(angle_steps) + angle
-        rs = np.clip(np.random.normal(avg_radius, spikeyness, n_vert), 0, 2 * avg_radius)
+        rs = np.clip(np.random.normal(avg_radius, spikeyness, n_vert), 0.1, 2 * avg_radius)
         xs = x_ctr + np.multiply(rs, np.cos(angles))
         ys = y_ctr + np.multiply(rs, np.sin(angles))
         points = [(x, y) for x, y in zip(xs, ys)]
 
         if len(points) == 1:
-            return Point(points)
+            geom = Point(points)
         elif len(points) == 2:
-            return LineString(points)
+            geom = LineString(points)
         else:
-            return Polygon(points)
+            geom = Polygon(points)
+
+        assert geom.is_valid, f"Generated geometry is invalid: {geom}"
+
+        return geom
```

Comment:
as-is
![image](https://github.com/TeamSPWK/cartesius/assets/97011129/536267a1-7598-4ee8-bb87-c9c42d74ea25)

to-be
![image](https://github.com/TeamSPWK/cartesius/assets/97011129/853b5161-e549-4d7d-b906-aed1171c5aa2)